sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/python/PythonMicroBatchStream.scala

```diff
@@ -66,9 +97,18 @@ class PythonMicroBatchStream(
   }
 
   override def stop(): Unit = {
+    cachedInputPartition.foreach(_._3.dropCache())
     runner.stop()
   }
 
   override def deserializeOffset(json: String): Offset = PythonStreamingSourceOffset(json)
 }
 
+object PythonMicroBatchStream {
+  var currentId = 0
```

Comment:
I don't think this is thread-safe, unless this variable is only accessible with nextStreamId. If you are not intentional to expose this to public, please explicitly block it. (private)

Reply:
Makes sense.